### DISCUSSION TITLE
metalama - WOW!!!

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/roslyn#62129] [General]
The Roslyn team should definitely take a look at Metalama by the postsharp folks. It is amazing!

Those features should really be incorporated straight into Roslyn.

**Comment by CyrusNajmabadi** [comment]
Can you give more information on what you would like to see?

**Reply by jmarolf** [reply]
My understanding is that Metalama is built using roslyn. If there are some apis or other things that would make building something like that easier, then I would love to know so we could improve those specific things. but for me the goal is to have roslyn be the basis for cool things the .NET community does. It doesn't need to do it all.

**Reply by Kry-a** [reply]
Definitely the Aspect feature of it, if you want examples you can go to https://try.metalama.net/
It makes having duplicated/boilerplate code much less common, having a feature like that built straight into Roslyn would be a life saver.

**Reply by PeterHevesi** [reply]
I would love, if you join Metalama communities:
The main one is on [Slack](https://metalama.slack.com/join/shared_invite/zt-137spabhz-6DOZXVTTrN5dNXljwVmSsg#/shared-invite/email)
But it's also on [Reddit](https://www.reddit.com/r/MetalamaFramework/)

**Reply by Xyncgas** [reply]
long ago I said maybe we can do code obfuscation in roslyn but many people said let's leave this to other companies
